### PR TITLE
Refactor RenderObject.toStringDeep and Widget.toStringDeep code to use DiagnosticsNode

### PR DESCRIPTION
**Pull request by jacob314**
Moved to a different pull request because the comments on the other CL were making github timeout.

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/switch.dart [hunk 1]
```diff
@@ -88,11 +88,16 @@ class CupertinoSwitch extends StatefulWidget {
...
+  void debugFillProperties(List<DiagnosticsNode> description) {
+    super.debugFillProperties(description);
+    description.add(new FlagProperty(
+      'value',
+      value: value,
+      ifTrue: 'on',
+      ifFalse: 'off',
+      showName: true,
+    ));
+    description.add(new ObjectFlagProperty<ValueChanged<bool>>('onChanged', onChanged, ifNull: 'disabled'));
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/switch.dart]: I would be more aggressive about putting these on one line. This one fits fine as far as I can tell, for example; in fact it's one character less than the one you do have on one line here:
```dart
    description.add(new FlagProperty('value', value: value, ifTrue: 'on', ifFalse: 'off', showName: true));
    description.add(new ObjectFlagProperty<ValueChanged<bool>>('onChanged', onChanged, ifNull: 'disabled'));
```

File: packages/flutter/lib/src/cupertino/switch.dart [hunk 2]
```diff
@@ -420,10 +425,22 @@ class _RenderCupertinoSwitch extends RenderConstrainedBox implements SemanticsAc
...
-    super.debugFillDescription(description);
-    description.add('value: ${value ? "checked" : "unchecked"}');
-    if (!isInteractive)
-      description.add('disabled');
+  void debugFillProperties(List<DiagnosticsNode> description) {
+    super.debugFillProperties(description);
+    description.add(new FlagProperty(
+      'value',
+      value: value,
+      ifTrue: "checked",
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/switch.dart]: nit: strings should use single quotes.

This also fits on one line fine:
```dart
    description.add(new FlagProperty('value', value: value, ifTrue: 'checked', ifFalse: 'unchecked', showName: true));
```

File: packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart [hunk 3]
```diff
@@ -6,6 +6,1584 @@ import 'package:meta/meta.dart';
...
+  ///
+  ///  * [Element], which uses this style.
+  dense,
+
+  /// Style that enables transitioning from nodes of one style to children of
+  /// another.
+  ///
+  /// See also:
+  ///
+  ///  * [RenderParagraph], which uses this style for
```
Comment (Hixie) [inline packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart]: incomplete sentence

File: packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart [hunk 4]
```diff
@@ -6,6 +6,1584 @@ import 'package:meta/meta.dart';
...
+  /// See [denseTextConfiguration] for an example of setting this to false.
+  final bool addBlankLineIfNoChildren;
+
+  /// Whether the name should be displayed on the same line as the description.
+  final bool isNameOnOwnLine;
+
+  /// Footer to add as its own line at the end of a non-root node.
+  ///
+  /// See [leafTextConfiguration] for an example of using footer to draw a box around
+  /// the node.  [footer] is indented the same amount as [prefixOtherLines].
```
Comment (Hixie) [inline packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart]: nitty nit nit: two spaces after period

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/cupertino/switch.dart (hunk 1 above)
Comment (jacob314) [inline packages/flutter/lib/src/cupertino/switch.dart]: done

File: packages/flutter/lib/src/cupertino/switch.dart (hunk 2 above)
Comment (jacob314) [inline packages/flutter/lib/src/cupertino/switch.dart]: done

File: packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart (hunk 3 above)
Comment (jacob314) [inline packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart]: done

File: packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart (hunk 4 above)
Comment (jacob314) [inline packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart]: done.

**Comment by zanderso** [comment]
This change is breaking the Fuchsia build. Is there any way it can be reverted and phased-in a different way?

**Comment by jacob314** [comment]
Unfortunately the change is breaking. Let me go ahead and create a CL that
fixes the Fuscia build break.

On Tue, Jul 25, 2017 at 10:45 AM, Zachary Anderson <notifications@github.com

**Comment by Hixie** [comment]
Build times on windows spiked with this patch, but it's not clear if that's a coincidence or not. Will monitor.

**Pull request merged**